AllReadyApp/Web-App/AllReady/Models/ApplicationUser.cs

```diff
@@ -17,5 +18,38 @@ public class ApplicationUser : IdentityUser
         public string TimeZoneId { get; set; }
 
         public string PendingNewEmail { get; set; }
+
+        public IEnumerable<ValidationResult> ValidateProfileCompleteness()
+        {
+            List<ValidationResult> validationResults = new List<ValidationResult>();
+
+            if (!EmailConfirmed)
+            {
+                validationResults.Add(new ValidationResult("Verify your email address", new string[] { nameof(Email) }));
+            }
+
+            if (string.IsNullOrWhiteSpace(Name))
+            {
+                validationResults.Add(new ValidationResult("Enter your name", new string[] { nameof(Name) }));
+            }
+
+            if (string.IsNullOrWhiteSpace(PhoneNumber))
+            {
+                validationResults.Add(new ValidationResult("Add a phone number", new string[] { nameof(PhoneNumber) }));
+            }
+
+            if (!string.IsNullOrWhiteSpace(PhoneNumber) && !PhoneNumberConfirmed)
```

Comment:
This can be an `else if (!PhoneNumberConfirmed)` attached to the if clause above.